beacon-chain/core/helpers/legacy.go

```diff
@@ -0,0 +1,28 @@
+package helpers
+
+import (
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/math"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
+)
+
+// IsLegacyDepositProcessPeriod determines if the current state should use the legacy deposit process.
+func IsLegacyDepositProcessPeriod(beaconState state.BeaconState, canonicalEth1Data *ethpb.Eth1Data) bool {
+	// Before the Electra upgrade, always use the legacy deposit process.
+	if beaconState.Version() < version.Electra {
+		return true
+	}
+
+	// Handle the transition period between the legacy and the new deposit process.
+	requestsStartIndex, err := beaconState.DepositRequestsStartIndex()
+	if err != nil {
+		// If we can't get the deposit requests start index,
+		// we should default to the legacy deposit process.
+		return true
+	}
+
+	//canonicalEth1Data should never be nil
+	eth1DepositIndexLimit := math.Min(canonicalEth1Data.DepositCount, requestsStartIndex)
```

Comment:
Lets align with the spec here:
https://github.com/ethereum/consensus-specs/pull/4106/files#diff-e19b9b12060b1ca6cb9d3d340e2c770d227a4c602fa7ec7781be400ba8662aa2R161